gdx/src/com/badlogic/gdx/assets/loaders/ShaderProgramLoader.java

```diff
@@ -20,88 +20,117 @@
 import com.badlogic.gdx.assets.AssetLoaderParameters;
 import com.badlogic.gdx.assets.AssetManager;
 import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.glutils.ShaderPart;
 import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.graphics.glutils.ShaderStage;
 import com.badlogic.gdx.utils.Array;
 import com.badlogic.gdx.utils.GdxRuntimeException;
 
-/** {@link AssetLoader} for {@link ShaderProgram} instances loaded from text files. If the file suffix is ".vert", it is assumed
- * to be a vertex shader, and a fragment shader is found using the same file name with a ".frag" suffix. And vice versa if the
- * file suffix is ".frag". These default suffixes can be changed in the ShaderProgramLoader constructor.
- * <p>
- * For all other file suffixes, the same file is used for both (and therefore should internally distinguish between the programs
- * using preprocessor directives and {@link ShaderProgram#prependVertexCode} and {@link ShaderProgram#prependFragmentCode}).
+/** {@link AssetLoader} for {@link ShaderProgram} instances loaded from text files. 
+ * Since asset loader only support one file reference, any of files can be used. 
```

Comment:
Could expand this explanation to say, "provided they use the same name prefix. If the names do not match, use ShaderProgramLoader to specify the individual names."